src/uri.rs

```diff
@@ -179,6 +175,39 @@ impl Uri {
     }
 }
 
+fn parse_scheme(s: &str) -> Option<usize> {
+    s.find(':')
+}
+
+fn parse_authority(s: &str) -> Option<usize> {
+    let v: Vec<&str> = s.split("://").collect();
+    let auth = v.last().unwrap()
```

Comment:
Good point! Admittedly, that was a copy-paste from the previous code and I didn't think about it. Pushed an update.